app/components/Track/index.js

```diff
@@ -103,6 +105,13 @@ export function Track({ track, currentTrackID }) {
     }
   };
 
+  const openTrackInfoPage = () => {
+    onTrackInfoClick([track]);
+    setTimeout(() => {
+      history.push(`/track/${trackId}`);
+    }, 0);
```

Comment:
why do we need a setTimeout?

Reply:
It is just there for good measure. Letting redux dispatch update then, render with new route.